server/src/test/java/org/elasticsearch/cluster/routing/allocation/ConcurrentRebalanceRoutingTests.java

```diff
@@ -32,18 +35,41 @@
 public class ConcurrentRebalanceRoutingTests extends ESAllocationTestCase {
 
     public void testClusterConcurrentRebalance() {
-        AllocationService strategy = createAllocationService(
-            Settings.builder()
-                .put("cluster.routing.allocation.node_concurrent_recoveries", 10)
-                .put("cluster.routing.allocation.cluster_concurrent_rebalance", 3)
-                .build()
-        );
+        boolean testFrozen = randomBoolean();
```

Comment:
it's probably worth having a separate method to test frozen? see https://github.com/elastic/elasticsearch/blob/main/TESTING.asciidoc#use-randomized-testing-for-coverage